outbound/index.js

```diff
@@ -313,29 +313,27 @@ exports.process_delivery = function (ok_paths, todo, hmails, cb) {
 };
 
 exports.build_todo = function (todo, ws, write_more) {
-    // Replacer function to exclude items from the queue file header
-    function exclude_from_json (key, value) {
-        switch (key) {
-            case 'message_stream':
-                return undefined;
-            default:
-                return value;
-        }
-    }
 
-    const todo_json = JSON.stringify(todo, exclude_from_json)
-    const todo_len = new Buffer.from(todo_json).length
+    const todo_str = `\n${JSON.stringify(todo, exclude_from_json)}\n`
+    const todo_len = Buffer.byteLength(todo_str)
 
-    const buf = new Buffer(4 + todo_len);
+    const buf = Buffer.alloc(4 + todo_len);
     buf.writeUInt32BE(todo_len, 0);
-    buf.write(todo_json, 4);
-
-    const continue_writing = ws.write(buf);
-    if (continue_writing) return process.nextTick(write_more);
+    buf.write(todo_str, 4);
 
-    ws.once('drain', write_more);
+    ws.write(buf, 'utf8', write_more);
```

Comment:
The semantics for `drain` are different to using the callback to write. This is an incorrect change. It's a very fringe condition where it happens, but using drain is the right thing to do (I've studied this code a lot to make sure it's correct).

Reply:
Do you know of any conditions where that's required? I'd love to get it covered by tests.

Reply:
See highWaterMark in https://nodejs.org/api/stream.html#stream_constructor_new_stream_writable_options

We can lower the value to configure the event.